app/forms/date_form.py

```diff
@@ -0,0 +1,136 @@
+import logging
+import calendar
+
+from wtforms import Form, FormField, SelectField, StringField
+from wtforms import validators
+
+from app.validation.validators import DateCheck, DateRangeCheck, DateRequired, MonthYearCheck
+
+logger = logging.getLogger(__name__)
+
+
+def get_date_form(answer=None, to_field_data=None, validate_range=False, error_messages=None):
+    """
+    Returns a date form metaclass with appropriate validators. Used in both date and
+    date range form creation.
+
+    :param error_messages: The messages during validation
+    :param answer: The answer on which to base this form
+    :param to_field_data: The data coming from the
+    :param validate_range: Whether the dateform should add a daterange validator
+    :return: The generated DateForm metaclass
+    """
+    class DateForm(Form):
+
+        MONTH_CHOICES = [('', 'Select month')] + [(str(x), calendar.month_name[x]) for x in range(1, 13)]
+
+        month = SelectField(choices=MONTH_CHOICES, default='')
+        year = StringField()
+
+    validate_with = [validators.optional()]
+
+    if not error_messages:
+        error_messages = {}
+
+    if answer['mandatory'] is True:
+        if 'validation' in answer and 'messages' in answer['validation'] \
```

Comment:
Yes, I think so. We don't want to change the validate_with validators unless it's mandatory....